packages/java-parser/src/tokens.js

```diff
@@ -39,7 +39,25 @@ FRAGMENT("HexDigit", "[0-9a-fA-F]");
 FRAGMENT("HexDigits", "{{HexDigit}}(({{HexDigit}}|'_')*{{HexDigit}})?");
 FRAGMENT("FloatTypeSuffix", "[fFdD]");
 FRAGMENT("LineTerminator", "(\\x0A|(\\x0D(\\x0A)?))");
-
+FRAGMENT("UnicodeMarker", "uu*");
+FRAGMENT("UnicodeEscape", "\\{{UnicodeMarker}}{{HexDigit}}{4}");
+FRAGMENT("RawInputCharacter", "\\{{UnicodeMarker}}[0-9a-fA-F]{4}");
+FRAGMENT("UnicodeInputCharacter", "[{{UnicodeEscape}}{{RawInputCharacter}}]");
+FRAGMENT("OctalDigit", "[0-7]");
+FRAGMENT("ZeroToThree", "[0-3]");
+FRAGMENT(
+  "OctalEscape",
+  "\\\\({{OctalDigit}}|{{ZeroToThree}}?{{OctalDigit}}{2})"
+);
+FRAGMENT("EscapeSequence", "\\\\[btnfr\"'\\\\]|{{OctalEscape}}");
+FRAGMENT(
+  "CharacterLiteral",
+  "'(?:[^\\\\']|(?:(?:{{EscapeSequence}})|\\\\u[0-9a-fA-F]{4}))'"
```

Comment:
are we strongly aligning with the spec style here or not?
This does not use the spec terminology.
https://docs.oracle.com/javase/specs/jls/se11/html/jls-3.html#jls-CharacterLiteral

Reply:
Do we not align with the spec because we use the [^] NOT group?

Reply:
I have reused the initial RegExp which was using the same [^] NOT group. I am trying to stay as close as possible to the spec while compromising parsing writing style with regexp writing style.